sdk/eventhub/azure-eventhub/samples/async_samples/recv_process_individual_checkpoint_by_time.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import asyncio
+import os
+import logging
+import time
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+STORAGE_CONNECTION_STR = os.environ["AZURE_STORAGE_CONN_STR"]
+BLOB_CONTAINER_NAME = "your-blob-container-name"  # Please make sure the blob container resource exists.
+
+logging.basicConfig(
+    level=logging.INFO,
+    format='%(threadName)s | %(asctime)s | %(name)-12s | %(levelname)-8s %(message)s', datefmt='%m-%d %H:%M:%S',)
+log = logging.getLogger(__name__)
+
+MAX_UPLOAD_DOC_TASK_COUNT = 100
+upload_doc_task_count = 0
```

Comment:
I would have expected something like a semaphore here...

Reply:
I changed to use semaphore lile 
```python
async def schedule_to_upload_doc(doc):
    await upload_doc_task_semaphore.acquire()
    asyncio.create_task(upload_doc(doc))

async def upload_doc(doc):
    await asyncio.sleep(1)  # http request to upload a doc
    upload_doc_task_semaphore.release()
```